packages/sdk/client-services/src/packlets/deprecated/system.ts

```diff
@@ -3,33 +3,47 @@
 //
 
 import { Config } from '@dxos/config';
-import { SystemService } from '@dxos/protocols/proto/dxos/client';
+import { Status, SystemService } from '@dxos/protocols/proto/dxos/client';
 
-import { ServiceContext } from '../services';
+import { ClientServicesHost } from '../services';
+import { VaultResourceManager } from '../vault';
 
 /**
  * @deprecated
  */
 export class SystemServiceImpl implements SystemService {
-  constructor(private readonly _config: Config, private readonly _serviceContext: ServiceContext) {}
+  private readonly _resourceManager?: VaultResourceManager;
+
+  constructor(private readonly _config: Config, private readonly _serviceHost: ClientServicesHost) {
+    if (this._serviceHost.compatibilityMode) {
```

Comment:
i'd change `compatibilityMode` to `exclusive` (uses locks to make sure only a single instance is running)

Reply:
or `lockKey`